stactools_aster/stactools/aster/stac.py

```diff
@@ -121,30 +128,30 @@ def create_item(xml_href: str,
     item.common_metadata.instruments = [ASTER_INSTRUMENT]
 
     # eo
-    item.ext.enable('eo')
-    item.ext.eo.cloud_cover = xml_metadata.cloud_cover
+    eo = EOExtension.ext(item)
```

Comment:
You'll have to add the extension to the item - the `.ext()` method doesn't automatically check to see if the extension is set. I had thought about this - having the `ext()` add the URI if it didn't already exist in the `stac_extensions` - but chose to avoid that check every time the method is invoked for performance, and also to be explicit about enabling it. 

You can do
```python
EOExtension.add_to(item)
eo = EOExtension.ext(item)
```

Also seeing how you expected it to work here makes me reflect on that API choice and whether it was the best. Any thoughts? cc @duckontheweb

Reply:
Yeah, I made a mistake here and didn't read the API completely, just looked at a couple tests (not to closely it seems!).

It can be hard to know in wheter `add_to` has been called previously in a given code path, so from a developer side I could see people just always using `add_to` then `ext` to protect themselves (e.g. if you're using an extension in a helper function). So I'd vote for `ext` to enable the extension if needed -- the check can't be that expense, right, just a dict lookup?

Reply:
Also very weird that none of the tests failed even though I never called `add_to` 🤔.

Reply:
Strictly speaking I think we should probably only treat a property as being part of an extension if the object explicitly implements that extension (i.e. has the URI in `stac_extensions`), so I think we should require developers to call `add_to` if they want to add the extension to the list. I'm worried that doing it behind the scenes in `ext` would mask a problem with an invalid object when people use PySTAC to read existing STAC objects, and it could also interfere with the odd case where someone is using their own custom fields that happen to overlap with an extension that they didn't intend to implement. I'm definitely open to pushback on any of these points, though.

I am thinking we should make 2 changes to PySTAC:

1) Raise an exception if the extension is not in `stac_extensions`. Right now it just returns any properties with the right name (e.g. `eo:cloud_cover`) whether the extension is actually implemented or not and it will also happily set those properties. I think @gadomski This is probably why it didn't error for you.
2) Ensure that `add_to` is idempotent. Right now it looks like it will append a URI as many times as the method is called. That way you could call it freely if you want to ensure that the extension is in `stac_extensions`

Reply:
Good call, I think that's a good path forward. And good catch with add_to not being idempotent, that's a bug.

Perhaps we could also add a flag to `ext` to add and get the extension in a one liner, e.g.

```python
eo  = EOExtensions.ext(item, add_if_missing=True)
```

Reply:
> Perhaps we could also add a flag to `ext` to add and get the extension in a one liner, e.g.
> 
> ```python
> eo  = EOExtensions.ext(item, add_if_missing=True)
> ```

Yeah, I like that idea!